format/checkpoint_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package format_test
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/trillian-examples/format"
+)
+
+func TestMarshal(t *testing.T) {
+	for _, test := range []struct {
+		c    format.Checkpoint
+		want string
+	}{
+		{
+			c: format.Checkpoint{
+				Ecosystem: "Log Checkpoint v0",
+				Size:      123,
+				RootHash:  []byte("bananas"),
+			},
+			want: "Log Checkpoint v0\n123\nYmFuYW5hcw==\n",
+		}, {
+			c: format.Checkpoint{
+				Ecosystem: "Banana Checkpoint v5",
+				Size:      9944,
+				RootHash:  []byte("the view from the tree tops is great!"),
+			},
+			want: "Banana Checkpoint v5\n9944\ndGhlIHZpZXcgZnJvbSB0aGUgdHJlZSB0b3BzIGlzIGdyZWF0IQ==\n",
+		},
+	} {
+		t.Run(string(test.c.RootHash), func(t *testing.T) {
+			got := test.c.Marshal()
+			if string(got) != test.want {
+				t.Fatalf("Marshal = %q, want %q", got, test.want)
+			}
+		})
+	}
+}
+
+func TestUnmarshalLogState(t *testing.T) {
+	for _, test := range []struct {
+		desc     string
+		m        string
+		want     format.Checkpoint
+		wantRest []byte
+		wantErr  bool
+	}{
+		{
+			desc: "valid one",
+			m:    "Log Checkpoint v0\n123\nYmFuYW5hcw==\n",
+			want: format.Checkpoint{
+				Ecosystem: "Log Checkpoint v0",
+				Size:      123,
+				RootHash:  []byte("bananas"),
+			},
+		}, {
+			desc: "valid with different ecosystem",
+			m:    "Banana Checkpoint v1\n9944\ndGhlIHZpZXcgZnJvbSB0aGUgdHJlZSB0b3BzIGlzIGdyZWF0IQ==\n",
+			want: format.Checkpoint{
+				Ecosystem: "Banana Checkpoint v1",
+				Size:      9944,
+				RootHash:  []byte("the view from the tree tops is great!"),
+			},
+		}, {
+			desc: "valid with trailing data",
```

Comment:
Maybe add one that also has a bunch of newlines in the trailing data. Perhaps even some blank lines (this format should be clear regardless of the current envelope we're planning to stick it in).

Reply:
Good call, done.